packages/remark-iframes/src/index.js

```diff
@@ -31,6 +31,18 @@ module.exports = function plugin (opts) {
     }
     return finalUrl
   }
+  function computeThumbnail (provider, url) {
+    let thumbnail = 'default image'
+    if (provider.thumbnail && provider.thumbnail.format) {
+      thumbnail = provider.thumbnail.format
+      Object.keys(provider.thumbnail).filter(key => key !== 'format')
+        .forEach(function (key) {
+          thumbnail = thumbnail.replace(/`{${key}}`/g,
+            new RegExp(provider.thumbnail[key]).exec(url)[1])
+        })
+    }
+    return thumbnail
+  }
```

Comment:
Replace logic isn't correct. You end up with a URL containing things like `/{id}/` which should have been replaced. This works:

```js
function computeThumbnail (provider, url) {
  let thumbnailURL = 'default image'
  if (provider.thumbnail && provider.thumbnail.format) {
    const thumbnailConfig = provider.thumbnail
    thumbnailURL = thumbnailConfig.format
    Object.keys(thumbnailConfig).filter(key => key !== 'format')
      .forEach((key) => {
        const search = new RegExp(`{${key}}`, 'g')
        const replace = new RegExp(thumbnailConfig[key]).exec(url)
        if (replace.length) thumbnailURL = thumbnailURL.replace(search, replace[1])
      })
  }
  return thumbnailURL
}
```

You can extract `computeThumbnail` and `computeFinalUrl` outside of the exported `plugin` function.